glean-core/src/common_metric_data.rs

```diff
@@ -30,15 +28,13 @@ impl Lifetime {
     }
 }
 
-impl TryFrom<i32> for Lifetime {
-    type Error = (); // FIXME: Proper error type needed
-
-    fn try_from(value: i32) -> Result<Lifetime, Self::Error> {
+impl From<i32> for Lifetime {
+    fn from(value: i32) -> Lifetime {
         match value {
-            0 => Ok(Lifetime::Ping),
-            1 => Ok(Lifetime::Application),
-            2 => Ok(Lifetime::User),
-            _ => Err(()),
+            0 => Lifetime::Ping,
+            1 => Lifetime::Application,
+            2 => Lifetime::User,
+            _ => Lifetime::Ping,
```

Comment:
Mh, what's the rationale behind this? In general, we try not to have "fallback" behaviours in Glean, but rather have clear erroring paths. What are we adjusting here?

Reply:
Hm, I see. 
I was trying to simplify the FFI component.
So far we did error report and still fallback to `Lifetime::Ping`.

I can see the use of "log error and bail out" though. Let me try something.

Reply:
Cheers! Thanks for 🐻 ing with me :P